lib/stormpath-sdk/resource/application.rb

```diff
@@ -98,10 +114,21 @@ def get_provider_account request
     Stormpath::Provider::AccountResolver.new(data_store).resolve_provider_account(href, request)
   end
 
+  def oauth_authenticate(options = {})
```

Comment:
I think this should just be called `authenticate` and we determine the type of authenticator that we need to use depending on the content of the request object. Also, I think people should be able to specify actual request objects (like `PasswordGrantRequest` or `RefreshGrantRequest`) instead of just an options array; this would be more object oriented and would produce a much cleaner code in my opinion.